Add MSC support for ESP32 S3

M-D-777 opened the pull request:
Hi, all

Any interest in adding MSC support for ESP32 S3?

The MSC feature is configurable with sdkconfig.usb_msc.

M-D-777 committed:
Add MSC support for ESP32 S3

PGNetHun commented on ports/esp32/modules/inisetup.py:
```diff
@@ -33,8 +33,10 @@ def fs_corrupted():
     print("Performing initial setup")
-    uos.VfsLfs2.mkfs(bdev)
-    vfs = uos.VfsLfs2(bdev)
+    # uos.VfsLfs2.mkfs(bdev)
```
Please do not change default `LittleFS` to `FAT`, as file is shared among all ESP32 variants (freeze in [manifest.py](https://github.com/micropython/micropython/blob/master/ports/esp32/boards/manifest.py#L1)).
Best way would be to have the `inisetup.py` file override in board folder with a custom `manifest.py`, just like in `UM_TINYS2` board's [manifest.py](https://github.com/micropython/micropython/blob/master/ports/esp32/boards/UM_TINYS2/manifest.py#L2) file

Red-M commented:
I've tested this patch and it works perfectly, however I can these changes would affect other boards, please limit the changes to only the S3 boards.

M-D-777 commented:
@PGNetHun @Red-M Thanks for your review. I will update a commit later.

M-D-777 committed:
add custom manifest.py for msc boot fatfs

M-D-777 commented:
Hi, @PGNetHun I have added custom manifest.py for msc boot fatfs.
Please review the newest commit, thanks.

M-D-777 committed:
Add usb msc feature to all of the ESP32 S2 and S3 variants.
By default, the usb feature is disabled by #boards/sdkconfig.usb_msc in mpconfigboard.cmake.

M-D-777 commented:
@Red-M @PGNetHun Hi, All. I have confirmed the USB MSC feature is compatible with ESP32 S2. So I add this feature to all of the ESP32 S2 and S3 variants.
By default, the usb feature is disabled by #boards/sdkconfig.usb_msc in mpconfigboard.cmake.
Do we need to enable the USB MSC feature by default, just like USB CDC?

M-D-777 committed:
Merge branch 'master' into feature/ESP32_S3_USB_MSC

# Conflicts:
#	ports/esp32/boards/GENERIC_S3_SPIRAM/mpconfigboard.cmake
